### PR TITLE
Baire Space

### PR DESCRIPTION
**Pull request by ElifUskuplu**
I've attempted to prove Baire space is uncountable.

To do this, I need to define another notion of countability called "directly countable". A type `X` is directly countable if there is a surjective map `f : ℕ → X`. I proved that if `X` is inhabitied, then `X` is countable iff `X` is directly countable. 

First, I proved Baire space is a  set.
Second, using direct countability and Lawvere's fixed point theorem,  I proved Baire space is uncountable.

Note that, as a trivial fact, I proved ` ℕ` is countable. As for equivalent definitions of countabiilty, I'm still working on it.

### CODE REVIEW

**Comment by ElifUskuplu** [comment]
I've tried to solve unsuccessful checks, but I couldn't get what is the problem.

**Comment by ElifUskuplu** [comment]
Btw, I have difficulty with working with `pre-commit`. I followed the instruction on the documentation, but when I run the command, I take the folowing output:

```
pre-commit 
trim trailing whitespace................................(no files to check)Skipped
fix end of files........................................(no files to check)Skipped
check yaml..............................................(no files to check)Skipped
check for added large files.............................(no files to check)Skipped
check for case conflicts................................(no files to check)Skipped
check for merge conflicts...............................(no files to check)Skipped
Python scripts formatting...............................(no files to check)Skipped
CSS, JS, YAML and Markdown (no codeblocks) formatting...(no files to check)Skipped
Remove extra blank lines in Agda files......................................Failed
- hook id: remove-extra-blank-lines
- duration: 0s
- exit code: 1

Executable `scripts/remove_extra_blank_lines.py` not found

Format Agda imports.........................................................Failed
- hook id: fix-imports
- duration: 0s
- exit code: 1

Executable `scripts/fix_imports.py` not found

Generate Agda namespace index modules.......................................Failed
- hook id: generate-namespace-index-modules
- duration: 0s
- exit code: 1

Executable `scripts/generate_namespace_index_modules.py` not found

Check the maximum line length is respected..............(no files to check)Skipped
- hook id: max-line-length
```
I also wonder what `pre-commit` checks? It seems no files is checked altough I edited two files.

**Comment by EgbertRijke** [comment]
Very nice PR!

`make pre-commit` needs to be run twice. The first time it will give errors, and the second time it should go through. It does a bunch of automatic formatting.

File: src/set-theory/countable-sets.lagda.md [hunk 1]
```diff
@@ -37,7 +48,8 @@ is-countable-Prop X = ∃-Prop (ℕ → Maybe (type-Set X)) is-surjective
 is-countable : {l : Level} → Set l → UU l
 is-countable X = type-Prop (is-countable-Prop X)
 
-is-prop-is-countable : {l : Level} (X : Set l) → is-prop (is-countable X)
+is-prop-is-countable : {l : Level} (X : Set l) →
```
Comment (EgbertRijke) [inline src/set-theory/countable-sets.lagda.md:51]: Our convention is that if we use line breaks, then we break the line at the first occasion. In this case, the line break should occur directly after `:`. I.e. 

```
is-prop-is-countable : 
  {l : Level} (X : Set l) → is-prop (is-countable X)
```

File: src/set-theory/countable-sets.lagda.md [hunk 2]
```diff
@@ -53,44 +65,85 @@ is-countable-Prop' X =
 is-countable' : {l : Level} → Set l → UU (lsuc lzero ⊔ l)
 is-countable' X = type-Prop (is-countable-Prop' X)
 
-is-prop-is-countable' : {l : Level} (X : Set l) → is-prop (is-countable' X)
+is-prop-is-countable' : {l : Level} (X : Set l) →
```
Comment (EgbertRijke) [inline src/set-theory/countable-sets.lagda.md:68]: Line break directly after `:`

File: src/set-theory/countable-sets.lagda.md [hunk 3]
```diff
@@ -53,44 +65,85 @@ is-countable-Prop' X =
...
-{-
-  is-countable-is-countable' :
-    is-countable' X → is-countable X
-  is-countable-is-countable' H =
+  is-directly-countable-is-countable : is-countable X →
+    is-directly-countable X
+  is-directly-countable-is-countable H =
+    apply-universal-property-trunc-Prop H
+      (is-directly-countable-Prop X)
+      λ P →
```
Comment (EgbertRijke) [inline src/set-theory/countable-sets.lagda.md:100]: We always write parentheses around lambda-expressions. Agda's "convention" that it allows parentheses of a lambda expression in the last argument to be omitted makes the code formatting look inconsistent.

File: src/set-theory/countable-sets.lagda.md [hunk 4]
```diff
@@ -53,44 +65,85 @@ is-countable-Prop' X =
...
-            -}
+          (pair
+            ( λ {
+              zero-ℕ → inr star ;
+              (succ-ℕ n) → inl ((shift-ℕ a (pr1 P)) n)})
+            ( λ {
+              (inl x) →
+                apply-universal-property-trunc-Prop (pr2 P x)
+                  (trunc-Prop (fib _ (inl x)))
+                  λ { (pair n p) →
```
Comment (EgbertRijke) [inline src/set-theory/countable-sets.lagda.md:128]: Parentheses around lambda-expression

File: src/set-theory/countable-sets.lagda.md [hunk 5]
```diff
@@ -53,44 +65,85 @@ is-countable-Prop' X =
...
+     is-surjective-f x = unit-trunc-Prop (pair (inl x) refl)
+
+  is-countable-is-directly-countable : is-directly-countable X
+    → is-countable X
+  is-countable-is-directly-countable H =
     apply-universal-property-trunc-Prop H
-      ( is-countable-Prop X)
-      ( λ (P , f) →
+      (is-countable-Prop X)
+      λ P →
```
Comment (EgbertRijke) [inline src/set-theory/countable-sets.lagda.md:118]: Parentheses around lambda-expression

File: src/set-theory/countable-sets.lagda.md (hunk 1 above)
Comment (ElifUskuplu) [reply src/set-theory/countable-sets.lagda.md]: Done!

File: src/set-theory/countable-sets.lagda.md (hunk 2 above)
Comment (ElifUskuplu) [reply src/set-theory/countable-sets.lagda.md]: Done!

File: src/set-theory/countable-sets.lagda.md (hunk 3 above)
Comment (ElifUskuplu) [reply src/set-theory/countable-sets.lagda.md]: Done!

File: src/set-theory/countable-sets.lagda.md (hunk 4 above)
Comment (ElifUskuplu) [reply src/set-theory/countable-sets.lagda.md:128]: Done!

File: src/set-theory/countable-sets.lagda.md (hunk 5 above)
Comment (ElifUskuplu) [reply src/set-theory/countable-sets.lagda.md]: Done!

**Comment by ElifUskuplu** [comment]
> Very nice PR!
> 
> `make pre-commit` needs to be run twice. The first time it will give errors, and the second time it should go through. It does a bunch of automatic formatting.

Got it. How can I solve the errors like ```Executable `scripts/remove_extra_blank_lines.py` not found``` ?

**Comment by EgbertRijke** [comment]
This error should not occur. I've reported it in #551 to ask for help.

File: src/set-theory/baire-space.lagda.md [hunk 6]
```diff
@@ -24,3 +42,45 @@ The Baire space is the type of functions `ℕ → ℕ`.
 baire-space : UU lzero
 baire-space = ℕ → ℕ
 ```
+
+## Properties
+
+### The Baire Space is a set.
+
+```agda
+is-baire-space-set : is-set baire-space
```
Comment (EgbertRijke) [inline src/set-theory/baire-space.lagda.md:51]: `is-set-bair-space`

File: src/set-theory/baire-space.lagda.md [hunk 7]
```diff
@@ -24,3 +42,45 @@ The Baire space is the type of functions `ℕ → ℕ`.
...
+
+baire-space-Set : Set lzero
+pr1 baire-space-Set = baire-space
+pr2 baire-space-Set = is-baire-space-set
+```
+
+### The Baire Space is uncountable.
+
+```agda
+is-baire-space-uncountable : is-uncountable baire-space-Set
```
Comment (EgbertRijke) [inline src/set-theory/baire-space.lagda.md:73]: `is-uncountable-baire-space`

File: src/set-theory/countable-sets.lagda.md [hunk 8]
```diff
@@ -53,44 +65,85 @@ is-countable-Prop' X =
...
 ```
 
 ## Properties
 
-### The two definitions of countability are equivalent
+If a set `X` is inhabited, then it is countable if and only if there is a
+surjective map `f : ℕ → X`. Let us call the latter as "directly countable".
 
 ```agda
+is-directly-countable-Prop : {l : Level} → Set l → Prop l
```
Comment (EgbertRijke) [inline src/set-theory/countable-sets.lagda.md:79]: The definitions `is-directly-countable-Prop`, `is-directly-countable` and `is-prop-is-directly-countable` can move up to the `## Definitions` section.

File: src/set-theory/baire-space.lagda.md (hunk 6 above)
Comment (ElifUskuplu) [reply src/set-theory/baire-space.lagda.md]: done!

File: src/set-theory/baire-space.lagda.md (hunk 7 above)
Comment (ElifUskuplu) [reply src/set-theory/baire-space.lagda.md]: done! Sorry, I'm forgetting this convention again and again :)

File: src/set-theory/countable-sets.lagda.md (hunk 8 above)
Comment (ElifUskuplu) [reply src/set-theory/countable-sets.lagda.md:79]: done!

**Pull request merged**